File: fair/example/assets/plugin/fair_common_plugin.js
let FairCommonPlugin = function () {
    return {
        http: function (resp) {
            fairCommonPluginRequest(resp, 'http');
        },
        requestPermission: function (resp) {
            fairCommonPluginRequest(resp, 'requestPermission');
        },
        selectPhoto: function (resp) {
            fairCommonPluginRequest(resp, 'selectPhoto');
        }                  
    }
}

File: fair/example/lib/src/page/plugins/fair_common_plugin.dart
import 'package:fair/fair.dart';
import 'package:fair_example/src/page/plugins/http.dart';
import 'package:fair_example/src/page/plugins/permission.dart';
import 'package:fair_example/src/page/plugins/photo.dart';

class FairCommonPlugin extends IFairPlugin
    with HttpPlugin, PermissionPlugin, PhotoPlugin {
  factory FairCommonPlugin() => _fairCommonPlugin;
  FairCommonPlugin._();
  static final FairCommonPlugin _fairCommonPlugin = FairCommonPlugin._();
  @override
  Map<String, Function> getRegisterMethods() {
    return <String, Function>{
      'http': http,
      'requestPermission': requestPermission,
      'selectPhoto': selectPhoto,
    };
  }
}
